electron/src/lib/CoreProtocol.ts

```diff
@@ -134,30 +134,27 @@ export class CustomProtocolHandler {
 
     ipcMain.on(EVENT_TYPE.ACTION.DEEP_LINK_SUBMIT, async (_event, url: string) => this.dispatchDeepLink(url));
 
-    if (platform.IS_MAC_OS) {
-      app.on('open-url', async (event, url) => {
-        event.preventDefault();
-        await this.dispatchDeepLink(url);
-      });
-    } else {
-      app.once('ready', async () => {
-        logger.info('App ready, looking for deep link in arguments ...');
-        const deepLink = this.findDeepLink(process.argv);
-        if (deepLink) {
-          await this.dispatchDeepLink(deepLink);
-        } else {
-          logger.info('No deep link found in arguments.');
-        }
-      });
-      app.on('second-instance', async (_event, argv) => {
-        logger.info('Second instance detected, looking for deep link in arguments ...');
-        const deepLink = this.findDeepLink(argv);
-        if (deepLink) {
-          await this.dispatchDeepLink(deepLink);
-        } else {
-          logger.info('No deep link found in arguments.');
-        }
-      });
-    }
+    app.on('open-url', async (event, url) => {
+      event.preventDefault();
+      await this.dispatchDeepLink(url);
+    });
+    app.once('ready', async () => {
+      logger.info('App ready, looking for deep link in arguments ...');
+      const deepLink = this.findDeepLink(process.argv);
+      if (deepLink) {
+        await this.dispatchDeepLink(deepLink);
+      } else {
+        logger.info('No deep link found in arguments.');
+      }
+    });
+    app.on('second-instance', async (_event, argv) => {
```

Comment:
Since currently this event is not registered for MacOS, Let's maybe prevent registering it for MacOS only. We don't really know what could be the side effect of it. 
WDYT?